pandas/_libs/window.pyx

```diff
@@ -1675,19 +1675,36 @@ def roll_generic(object obj,
     return output
 
 
-def roll_window(ndarray[float64_t, ndim=1, cast=True] values,
-                ndarray[float64_t, ndim=1, cast=True] weights,
-                int minp, bint avg=True):
+# ----------------------------------------------------------------------
+# Rolling sum and mean for weighted window
+
+
+def roll_weighted_sum(ndarray[float64_t, ndim=1, cast=True] values,
+                      ndarray[float64_t, ndim=1, cast=True] weights,
+                      int minp):
+    return _roll_weighted_sum_mean(values, weights, minp, avg=0)
+
+
+def roll_weighted_mean(ndarray[float64_t, ndim=1, cast=True] values,
+                       ndarray[float64_t, ndim=1, cast=True] weights,
+                       int minp):
+    return _roll_weighted_sum_mean(values, weights, minp, avg=1)
+
+
+def _roll_weighted_sum_mean(ndarray[float64_t, ndim=1, cast=True] values,
+                            ndarray[float64_t, ndim=1, cast=True] weights,
+                            int minp, bint avg=True):
     """
     Assume len(weights) << len(values)
     """
     cdef:
-        float64_t[:] output, tot_wgt, counts
+        ndarray[float64_t] output, tot_wgt, counts
```

Comment:
shouldn't these *all* take ``float64_t[:]`` ? rather than using the cast form of ndarray (which we don't use much)

Reply:
are you pointing args with `cast=True`? could change arguments, `tot_wgts` and `counts`. should we also do that for `output` as it should be converted from memory view?

Reply:
yes ; we ideally should have all args as memory views (unless we actually need a ndarray which it’s better to explicitly call np.asarray inside the code)

Reply:
used np.asarray [here](https://github.com/pandas-dev/pandas/pull/27403/commits/68d2b16028ee3f450ac8d42b51e7b10a2e2e2b0a#diff-813747abc5dedd9f437f30f692c84f48R452). we could change all other `ndarray[...]` to mem views in window.pyx